server/server.go

```diff
@@ -219,15 +220,13 @@ func handleHealth(w http.ResponseWriter, r *http.Request) {
 }
 
 func handleWebsocket(w http.ResponseWriter, r *http.Request) {
-	userId := getUserIdFromCookie(r)
-	responseHeader := setUserIdToCookie(w, userId)
-
-	if len(userId) == 0 {
-		userId = NewUserId()
-		responseHeader = setUserIdToCookie(w, userId)
+	userId := NewUserId()
+	if cookie, err := r.Cookie(cookiename); err == nil {
+		userId = UserId(cookie.Value)
 	}
+	http.SetCookie(w, mintCookie(userId))
 
-	ws, err := upgrader.Upgrade(w, r, responseHeader)
+	ws, err := upgrader.Upgrade(w, r, w.Header())
```

Comment:
I believe the last argument here can be `nil`, since we haven't done anything with `w`. Is that right?

Reply:
We have indeed done something with `w`! We have mutated it by adding a "Set-Cookie" header. I think you're right, though. A quick test confirms the upgrader sends the appropriate "Set-Cookie" header when the third argument is `nil`. I'll fix that.